File: src/main/java/pl/dmichalski/algorithms/_4_sum_zero/SumToZeroService1.java
package pl.dmichalski.algorithms._4_sum_zero;

public class SumToZeroService1 {

    /**
     * O(n2) complexity
     */
    public int[] sumZero(int[] values) {
        for (int i = 0; i < values.length; i++) {
            for (int j = i + 1; j < values.length; j++) {
                if (values[i] + values[j] == 0) {
                    return new int[]{values[i], values[j]};
                }
            }
        }

        return null;
    }

}

File: src/main/kotlin/pl/dmichalski/algorithms/algorithms/_04_sum_zero/SumToZeroService1.kt
package pl.dmichalski.algorithms.algorithms._04_sum_zero

internal class SumToZeroService1 {

    /**
     * O(n2) time complexity
     */
    fun sumZero(values: IntArray): IntArray? {
        for (i in values.indices) {
            for (j in i + 1 until values.size) {
                if (values[i] + values[j] == 0) {
                    return intArrayOf(values[i], values[j])
                }
            }
        }
        return null
    }

}
